src/UserProfileService.Sync/Services/SynchronizationService.cs

```diff
@@ -202,13 +209,21 @@ public async Task<ProcessDetail> GetDetailedProcessAsync(Guid id, CancellationTo
                 };
         }
 
-        if (isLockReleased)
+        if (isLockReleased) // process is running longer than 15 min (default value)
         {
-            _logger.LogInfoMessage("Sync lock object is released, sync status will be send ", LogHelpers.Arguments());
+            _logger.LogInfoMessage("Sync lock object is released, sync status will be returned ", LogHelpers.Arguments());
 
-            foreach (ProcessView processView in processViews)
+            if (processViews.Any())
             {
-                await DeclareProcessAbortedAsync(processView.CorrelationId, cancellationToken);
+                bool areProcessesHanging = processViews.Any(
+                    p => p.LastActivity != null
+                         && DateTime.Compare(DateTime.UtcNow, p.LastActivity.Value.AddMinutes(_syncConfiguration.DelayBeforeTimeoutForStep)) > 0);
```

Comment:
Is there another value to check if `LastActivity` is `null`? Like a creation time. Or are we safe to assume that `LastActivity` will at least be set once when initialized?

Reply:
LastActivity is set by process initialization